Fix create, push and watch commands on windows

dgolovin opened the pull request:
Fix #641 and #635

surajnarwade commented on pkg/component/component.go:
```diff
@@ -131,6 +132,25 @@ func CreateFromPath(client *occlient.Client, name string, componentImageType str
 }
 
+// Reads file path form URL file:///C:/path/to/file to C:\path\to\file
+func ReadFilePath(u *url.URL) string {
```
this functions  `ReadFilePath` and `GenFilePath` should go in `pkg/util/util.go`

surajnarwade replied:
this functions `ReadFilePath` and `GenFilePath` should go in `pkg/util/util.go`

surajnarwade commented:
other than one comment, everything looks good to me, worked on my linux machine
Thanks @dgolovin :)

dgolovin commented:
@surajnarwade thanks for the review. I've moved those methods to `util` package.

dgolovin commented:
@kadel could you review it or assign someone?

surajnarwade commented:
cc @anmolbabu @mik-dass @syamgk please review this PR

mik-dass commented:
@dgolovin @surajnarwade I will test this on a windows machine

cdrage commented on pkg/occlient/occlient.go:
```diff
@@ -1647,7 +1647,7 @@ func tar(tw *taro.Writer, fileName string, destFile string) error {
 	splitFileName := strings.Split(fileName, destFile)[1]
 
-	hdr.Name = destFile + splitFileName
+	hdr.Name = destFile + strings.Replace(splitFileName, "\\", "/", -1)
```
Mind adding a small comment that this is intended for Windows / Mac OS / Linux support?

cdrage commented on pkg/util/util.go:
```diff
@@ -59,3 +62,22 @@ func NamespaceOpenShiftObject(componentName string, applicationName string) (str
 }
+
+// Reads file path form URL file:///C:/path/to/file to C:\path\to\file
+func ReadFilePath(u *url.URL) string {
```
We should add tests for this.. but we can ignore this for now just so we can fix the commands on Windows for the time being. Can you add a TODO: Write tests comment?

dgolovin replied:
sure, I'll look if I can add tests, to make PR checks back to green, if I fail, then adding `TODO` is my backup plan.

anmolbabu commented:
@dgolovin does this PR fix https://github.com/redhat-developer/odo/issues/639 also ?

dgolovin commented:
@anmolbabu I think #639 is a different problem and this fix is not going to help.

dgolovin commented:
@cdrage I've added tests and comment, why replacing '\' to '/' is required.

dgolovin commented:
I've squashed all commits in one and fixed some typos in test messages. 
@cdrage @anmolbabu @mik-dass @surajnarwade  I have no rights to push, could you release it to the master branch?

mik-dass commented on pkg/util/util.go:
```diff
@@ -59,3 +61,24 @@ func NamespaceOpenShiftObject(componentName string, applicationName string) (str
+
+const WIN = "windows"
+
+// Reads file path form URL file:///C:/path/to/file to C:\path\to\file
```
Shouldn't it be C:\\path\\to\\file?

mik-dass replied:
Shouldn't it be C:\\\path\\\to\\\file ?

dgolovin commented:
@mik-dass fixed. Thanks!

cdrage commented:
Unfortunately this needs a re-based as we merged in some pretty big PRs. The code LGTM however! So after re-base, feel free to merge.

dgolovin commented:
@cdrage rebased on top of master.
